sklearn/tests/test_kernel_approximation.py

```diff
@@ -114,7 +114,26 @@ def test_additive_chi2_sampler():
     Y_neg[0, 0] = -1
     msg = "Negative values in data passed to"
     with pytest.raises(ValueError, match=msg):
-        transform.transform(Y_neg)
+        transform.fit(X)
+
+    # test that the sample interval is set correctly
+    sample_steps_available = [1, 2, 3]
+    for sample_steps in sample_steps_available:
+
+        # test that the sample_interval is initialized correctly
+        transform = AdditiveChi2Sampler(sample_steps=sample_steps)
+        assert transform.sample_interval is None
+
+        # test that the sample_interval is changed in the fit method
+        transform.fit(X)
+        assert transform.sample_interval_ is not None
```

Comment:
I would avoid this check since it will raise a warning.
We can check this in the test together with the warning.